language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/components/helpers/TemplateUtils.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava.components.helpers;
+
+import com.daml.ledger.javaapi.data.Identifier;
+import com.daml.ledger.javaapi.data.Record;
+import com.daml.ledger.javaapi.data.Template;
+import com.daml.ledger.javaapi.data.Value;
+import com.daml.ledger.rxjava.components.helpers.CreatedContract;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Function;
+
+public class TemplateUtils {
+
+    /**
+     * Creates a transform function which is aware of the template types. Useful to be passed to
+     * Bot.wire.
+     *
+     * @param allowedTemplates the list of allowed template types
+     * @return a transformation function returning a new contract typed as one of the allowed
+     *     templates
+     */
+
+    public static Function<CreatedContract, Template> contractTransformator(
+            Class<? extends Template>... allowedTemplates) {
+        return createdContract -> {
+            Record args = createdContract.getCreateArguments();
+            for (Class template : allowedTemplates) {
```

Comment:
This way every `CreatedContract` will result in looping through the `allowedTemplates` and running the reflection.
You can do something like:

```java
public static Function<CreatedContract, Template> contractTransformator(
            Class<? extends Template>... allowedTemplates) {
  Map<Identifier, Method>factories = ...;
  for (Class template : allowedTemplates) {
    // do the reflection, and store the "fromValue" Method object as value for the "Identifier" key
  }
  return createdContract -> {
    if (!factories.contains(createdContract.getTemplateId())) { throw new .... }
    else return factories.get(createdContract.getTemplateId()).invoke(null, args)
  };
```

Reply:
Thanks, great idea, I changed this.